src/utils/tryGetDefaultDockerContext.ts

```diff
@@ -0,0 +1,60 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as cp from 'child_process';
+import { DockerOptions } from 'dockerode';
+
+const unix = 'unix://';
+const npipe = 'npipe://';
+
+// Not exhaustive--only the properties we're interested in
+interface IDockerEndpoint {
+    Host?: string;
+}
+
+// Also not exhaustive--only the properties we're interested in
+interface IDockerContext {
+    Endpoints: { [key: string]: IDockerEndpoint }
+}
+
+let cachedDockerContext: DockerOptions;
+let cachedDockerContextTimestamp: number;
+
+export function tryGetDefaultDockerContext(): DockerOptions {
+    // First check the cached value--we will keep result from last 10 seconds
+    if (cachedDockerContextTimestamp && cachedDockerContextTimestamp + 10000 > Date.now()) {

```

Comment:
Any particular reason for holding onto the cached context for only 10s?

Reply:
We can make it longer (or shorter), but @Chuxel used 10s in his corresponding code for the Remote extension.

Making it longer would primarily benefit tests. The tests do `refreshDockerode` a lot, and they do take longer than 10s total, so they usually end up calling this 2-3 times.

Making it shorter would be bad for tests but better for users, if they are quickly changing between different settings.

Reply:
I changed the settings watcher to only call `refreshDockerode` on relevant setting changes (which the tests do not do), so caching no longer matters for tests. I removed caching entirely since it could have adverse effects when users are changing things in the settings page.